### PR TITLE
Iam auth serverless

### PR DESCRIPTION
**Pull request by ivanfslee**
Change dbClient.js and model.js to return functions that take parameters to enable IAM auth token to be passed in via http headers.

### CODE REVIEW

**Review by ChaseSobey — APPROVED** [review]
nice changes, we really needed this functionality working and you got a lot of the issues ironed out in this merge!

File: src/api/index.js [hunk 1]
```diff
@@ -9,18 +9,32 @@ import getSecret from '../utils/rds/getSecret.js';
...
+import Secrets from '../utils/rds/model.js';
 
 const app = express();
 
 app.use(cors());
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
+app.use((req, res, next) => {
+    console.log('req.headers is: ', req.headers);
```
Comment (taronavagyan) [inline src/api/index.js:22]: This should be deleted before production.

**Review by taronavagyan — COMMENTED** [review]
Looks good, nice work!

**Review by taronavagyan — APPROVED** [review]

File: src/api/index.js (hunk 1 above)
Comment (ivanfslee) [reply src/api/index.js:22]: I'll make a note of that for next commit.

**Pull request merged**